### PR TITLE
Add support for testing boolean expressions

### PR DESCRIPTION
**Pull request by amorimjuliana**
Evaluating boolean expressions is very useful for creating audiences since the goal is only to check if a user fulfills some requirements.

## Summary

Add support for testing boolean expressions.

Evaluating boolean expressions is very useful for creating audiences since the goal is only to check if a user fulfills some requirements.

### Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: src/plug.ts [hunk 1]
```diff
@@ -280,6 +280,12 @@ export class GlobalPlug implements Plug {
         return this.sdk.evaluate(expression, options);
     }
 
+    public test(expression: string, options: EvaluationOptions = {}): Promise<boolean> {
+        return this.sdk.evaluate(expression, options)
+            .then(result => result === true)
+            .catch(() => false);
```
Comment (Fryuni) [inline src/plug.ts:286]: I get the reasoning here of treating runtime evaluation errors as false, like `user's preference's shopPattern is "eager"`.
But I don't think we should hide authentication errors at the plug level, those should be propagated and it's the user's responsibility to handle them

Comment (amorimjuliana) [reply src/plug.ts:286]: Agreed :)

File: src/plug.ts [hunk 2]
```diff
@@ -280,6 +280,11 @@ export class GlobalPlug implements Plug {
         return this.sdk.evaluate(expression, options);
     }
 
+    public test(expression: string, options: EvaluationOptions = {}): Promise<boolean> {
+        return this.sdk.evaluate(expression, options)
+            .then(result => result === true);
```
Comment (marcospassos) [inline src/plug.ts:285]: Why not `this.evaluate()`?

File: test/plug.test.ts [hunk 3]
```diff
@@ -602,6 +602,67 @@ describe('The Croct plug', () => {
...
+        const evaluate = jest.spyOn(sdkFacade, 'evaluate').mockRejectedValue(undefined);
+
+        const promise = croct.test('user\'s name is "Carol"', {timeout: 5});
+
+        await expect(promise).rejects.toBeUndefined();
+
+        expect(evaluate).toBeCalledWith('user\'s name is "Carol"', {timeout: 5});
+    });
+
+    test('should not allow to test expressions if unplugged', () => {
```
Comment (marcospassos) [inline test/plug.test.ts:662]: Is this really needed?

**Pull request merged**